libraries/EEPROM/examples/eeprom_extra/eeprom_extra.ino

```diff
@@ -20,68 +20,117 @@ void setup() {
     ESP.restart();
   }
 
-  int address = 0;                                  // Same address is used through the example
+  int address = 0;
 
   EEPROM.writeByte(address, -128);                  // -2^7
-  Serial.println(EEPROM.readByte(address));
+  address += sizeof(byte);
 
   EEPROM.writeChar(address, 'A');                   // Same as writyByte and readByte
-  Serial.println(char(EEPROM.readChar(address)));
+  address += sizeof(char);
 
   EEPROM.writeUChar(address, 255);                  // 2^8 - 1
-  Serial.println(EEPROM.readUChar(address));
+  address += sizeof(unsigned char);
 
   EEPROM.writeShort(address, -32768);               // -2^15
-  Serial.println(EEPROM.readShort(address));
+  address += sizeof(short);
 
   EEPROM.writeUShort(address, 65535);               // 2^16 - 1
-  Serial.println(EEPROM.readUShort(address));
+  address += sizeof(unsigned short);
 
   EEPROM.writeInt(address, -2147483648);            // -2^31
-  Serial.println(EEPROM.readInt(address));
+  address += sizeof(int);
 
   EEPROM.writeUInt(address, 4294967295);            // 2^32 - 1
-  Serial.println(EEPROM.readUInt(address));
+  address += sizeof(unsigned int);
 
   EEPROM.writeLong(address, -2147483648);           // Same as writeInt and readInt
-  Serial.println(EEPROM.readLong(address));
+  address += sizeof(long);
 
   EEPROM.writeULong(address, 4294967295);           // Same as writeUInt and readUInt
-  Serial.println(EEPROM.readULong(address));
+  address += sizeof(unsigned long);
 
   int64_t value = -9223372036854775808;             // -2^63
   EEPROM.writeLong64(address, value);
-  value = 0;                                        // Clear value
+  address += sizeof(int64_t);
+
+  uint64_t  Value = 18446744073709551615;           // 2^64 - 1
+  EEPROM.writeULong64(address, Value);
+  address += sizeof(uint64_t);
+
+  EEPROM.writeFloat(address, 1234.1234);
+  address += sizeof(float);
+
+  EEPROM.writeDouble(address, 123456789.123456789);
+  address += sizeof(double);
+
+  EEPROM.writeBool(address, true);
+  address += sizeof(bool);
+
+  String sentence = "I love ESP32.";
+  EEPROM.writeString(address, sentence);
+  address += sentence.length();
+
+  char gratitude[21] = "Thank You Espressif!";
+  EEPROM.writeString(address, gratitude);
+  address += 21;
+
+  // See also the general purpose writeBytes() and readBytes() for BLOB in EEPROM library
+  EEPROM.commit();
+  address = 0;
+  
+  Serial.println(EEPROM.readByte(address));
+  address += sizeof(byte);
+
+  Serial.println(char(EEPROM.readChar(address)));
```

Comment:
Does this actually compile?  Shouldn't it be `printlin((char)EEPROM.readChar(address));`

Reply:
travis would have complained :) not sure how Make will like it though